lib/core/MantraModel.js

```diff
@@ -1,3 +1,25 @@
+/**
+ * Object with the properties needed to instantiate a MantraModel class.
+ * @typedef {Object} MantraModelConfig
+ * @property {Array} fields - Fields to resolve from a MantraSchema.
+ * @property {String} name - Name identifying the model to be set on its parent class.
+ */
+
+/**
+ * Entity schema configuration set at installation.
+ * @typedef {Object} MantraSchema
+ * @property {Object} fields - Fields to serve as source for Model's fields.
+ * @property {Object} model - Object with MantraModelConfigs to pick the base one for the class.
+ */
+
+/**
+ * MantraModel class constructor params.
+ * @typedef {Object} ConstructorParams
```

Comment:
I think that `ConstructorParams` it's a very generic name since this `typedef` addresses the MantraModel constructor. Thus, I would suggest changing the `ConstructorParams` to `MantraModelParams`.

`MantraModelParams`: It indicates that there's a resource called "MantraModel" and that the interface detail the "Params" for that resource.

The suggestion is the following:
```suggestion
 * @typedef {Object} MantraModelParams
```

If the idea was that the `typedef` call `ConstructorParams` was going to be re-used by constructors of different classes (because the properties passed are always the same) then I would suggest defining as generic as possible every property as well as the `typedef` description.

```suggestion
 * Base params for constructors of Mantra classes
 * @typedef {Object} ConstructorParams
 * @property {datatype1} genericProp1 - [DEFINITION OF THE COMMON PROP 1]
  * @property {datatype2} genericProp2 - [DEFINITION OF THE COMMON PROP 2]
```